components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2540,10 +2571,28 @@ private static Certificate getCertificate(String tenantDomain, int tenantId) thr
                 // derive key store name
                 String ksName = tenantDomain.trim().replace(".", "-");
                 String jksName = ksName + ".jks";
-                keyStore = tenantKSM.getKeyStore(jksName);
-                publicCert = keyStore.getCertificate(tenantDomain);
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Loading default tenant certificate for tenant : %s from the KeyStore" +
+                            " %s", tenantDomain, ksName));
+                }
+                try {
+                    keyStore = tenantKSM.getKeyStore(jksName);
+                    publicCert = keyStore.getCertificate(tenantDomain);
+                } catch (KeyStoreException e) {
+                    throw new IdentityOAuth2Exception("Error occurred while loading public certificate for tenant: " +
+                            tenantDomain, e);
+                } catch (Exception e) {
+                    throw new IdentityOAuth2Exception("Error occurred while loading Keystore: " + keyStore + " for " +
```

Comment:
Is passing the Keystore reference useful here in the exception message?

Reply:
Since the method is about getting the certificate from the tenant domain, the Keystore reference calculated using that value does not seems useful. Therefore I will remove it.

Reply:
fixed